lib/modules/manager/terraform/lockfile/index.ts

```diff
@@ -177,6 +177,18 @@ export async function updateArtifacts({
         if (!updateLock) {
           continue;
         }
+        if (dep.isLockfileUpdate) {
```

Comment:
Please move that logic into getNewConstraint. 

We have their already logic to handle such things. I guess something is not working their correctly

Reply:
hmm I looked into that and it does not make to me to do so, this condition is aborting the update to the `newVersion`. Moving it into `getNewConstraint` would only allow to return the `oldConstraint`, Renovate would still proceed with updating the version

Reply:
we should add a break if old and new constraints are equal

Reply:
`break` this loop in this condition instead of `continue`?

https://github.com/renovatebot/renovate/blob/cc64f8dabd414390ff0540c5c9785e837e2d3142/lib/modules/manager/terraform/lockfile/index.ts#L166

Other providers have their own constraints, I think we want to check those independently, so continuing seems right

Reply:
you're right, we should continue with next dep